tsdb/tombstones/tombstones.go

```diff
@@ -145,6 +138,44 @@ func WriteFile(logger log.Logger, dir string, tr Reader) (int64, error) {
 	return int64(size), fileutil.Replace(tmp, path)
 }
 
+func Encode(tr Reader) ([]byte, error) {
+	buf := encoding.Encbuf{}
+	buf.PutByte(tombstoneFormatV1)
+	err := tr.Iter(func(ref uint64, ivs Intervals) error {
+		for _, iv := range ivs {
+			buf.PutUvarint64(ref)
+			buf.PutVarint64(iv.Mint)
+			buf.PutVarint64(iv.Maxt)
+		}
+		return nil
+	})
+	return buf.Get(), errors.Wrap(err, "encoding tombstones")
+}
+
+func Decode(b []byte) (Reader, error) {
+	d := &encoding.Decbuf{B: b}
```

Comment:
I don't really like the fact that we expect very specific bytes. It has to be WITHOUT magic number. Simarly hash has to be added by someone else after Encode and has to remember to avoid format byte. Too many things to remember, to easy to forget. Can we embed those fragile details into Encode & Decode? (:

Reply:
Take for example marshaling of YAML, you would expect just the plain encoding of the YAML and won't expect checksum to be added or some magic number :). Magic number is something specific to the file and not the encoding of tombstones. Similarly the checksum is what the caller decides to add and should not be part of the encoding. 

So I prefer keeping this encode and decode method only for plain encoding and decoding of tombstones. When using this for snapshot, I don't want checksum or magic number there, because magic number does not make sense there and checksum is added by WAL package.

Reply:
fair point.